packages/components/toaster/src/__tests__/Toaster.spec.tsx

```diff
@@ -0,0 +1,169 @@
+    );
+
+    expect(screen.getByText('Success Message')).toBeInTheDocument();
+    const toaster = screen.getByText('Success Message').closest('div');
+    expect(toaster).toHaveStyle(
+      'bottom: 1rem; left: 50%; transform: translateX(-50%);',
+    );
+  });
+  it('should disappear after the specified duration', async () => {
+    jest.useFakeTimers();
+
+    waitFor(() => {
+      render(
+        <Toaster typeToaster="success" position="top-right" duration={1000}>
+          Temporary message
+        </Toaster>,
+      );
+    });
+
+    expect(screen.getByText('Temporary message')).toBeInTheDocument();
+
+    jest.advanceTimersByTime(1000);
+
+    await waitFor(async () => {
+      expect(screen.queryByText('Temporary message')).not.toBeInTheDocument();
+    });
+
+    jest.useRealTimers();
+  });
+  it('should disappear when CloseButton is clicked', async () => {
+    const ToasterTest = () => {
+      return (
+        <div>
+          <Toaster
+            typeToaster="warning"
+            position="top-right"
+            showCloseButton={true}
+          >
+            Warning Message
+          </Toaster>
+        </div>
+      );
+    };
+
+    render(<ToasterTest />);
+    expect(screen.getByText('Warning Message')).toBeInTheDocument();
+    fireEvent.click(screen.getByLabelText('Close toaster'));
+    await waitFor(() => {
+      expect(screen.queryByText('Warning Message')).not.toBeInTheDocument();
+    });
+  });
+  it('should shows toaster when button is clicked', () => {
+    const ToasterTest = () => {
+      const [showToaster, setShowToaster] = useState(false);
+
+      return (
+        <div>
+          <Button onClick={() => setShowToaster(true)}>Show Toaster</Button>
+          {showToaster && (
+            <Toaster typeToaster="info" position="top-center">
+              Click triggered
+            </Toaster>
+          )}
+        </div>
+      );
+    };
+    render(<ToasterTest />);
+    expect(screen.queryByText('Click triggered')).not.toBeInTheDocument();
+    fireEvent.click(screen.getByText('Show Toaster'));
+    expect(screen.getByText('Click triggered')).toBeInTheDocument();
+  });
+  it('should display a progress bar that decreases and toaster should close after the specified duration', async () => {
+    jest.useFakeTimers();
+
+    render(
+      <Toaster
+        typeToaster="info"
+        position="top-center"
+        duration={2000}
+        showProgressBar={true}
+      >
+        Progress Bar Message
+      </Toaster>,
+    );
+
+    expect(screen.getByText('Progress Bar Message')).toBeInTheDocument();
+    const progressBar = screen.getByRole('progressbar');
+    expect(progressBar).toBeInTheDocument();
+
+    jest.advanceTimersByTime(1000);
+    expect(progressBar).toBeInTheDocument();
+
+    jest.advanceTimersByTime(1000);
+    await waitFor(() => {
+      expect(
+        screen.queryByText('Progress Bar Message'),
+      ).not.toBeInTheDocument();
+    });
+
+    jest.useRealTimers();
```

Comment:
Why 'useRealTimers' at the end?

Reply:
When I was reading about testing with useFakeTimers(), I realized that is a good practice to prevent interference with other tests that may occur afterwards. That's why I used it. Do you think it's unnecessary?

Reply:
https://jestjs.io/docs/next/timer-mocks

useFakeTimers is typically set outside of each test. I don't see the need here.

Reply:
I analyzed the test and found it to be redundant, so I decided to remove it.